### PR TITLE
Migration guide for deprecations removed after v3.13

### PR DESCRIPTION
**Pull request by Piinks**
_Description of what this PR is changing or adding, and why:_

This adds the full migration guide for all deprecated APIs that were removed from the framework after the 3.13 stable release.

_Issues fixed by this PR (if any):_

Fixes https://github.com/flutter/flutter/issues/133171

## Presubmit checklist

- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

File: src/release/breaking-changes/3-13-deprecations.md
Comment (atsansone) [inline src/release/breaking-changes/3-13-deprecations.md:5]: **chore:** The [build error](https://github.com/flutter/website/actions/runs/6190333187/job/16806357669?pr=9394#step:3:1136) exists because this YAML value needs to be indented.
```suggestion
description: >
  After reaching end of life, the following deprecated APIs
  were removed from Flutter.
```

File: src/release/breaking-changes/3-13-deprecations.md
Comment (atsansone) [inline src/release/breaking-changes/3-13-deprecations.md:380]: **issue (blocking):** This reads with some difficulty.
```suggestion
To prepare for multi-window support,
many deprecated properties of `TestWindow` have been removed.
While `TestWindow` has been deprecated, it does not qualify
for removal at this time.
Migrating the expired properties now will help in migrating from `TestWindow`.
```

File: src/release/breaking-changes/3-13-deprecations.md
Comment (atsansone) [inline src/release/breaking-changes/3-13-deprecations.md:406]: **suggestion (non-blocking):** Consider a more concise phrasing. (Also update the link to match.)
```suggestion
To learn more about this `TestWindow` update, check out
[`TestWindow` migration guide][].

[`TestWindow` migration guide]: {{site.url}}/release/breaking-changes/window-singleton
```

**Review by atsansone — APPROVED** [review]
LGTM % some clean up (line breaks) and suggested copy changes. Two blockers exist on [lines 3 to 5](https://github.com/flutter/website/pull/9394/files#diff-afe162e1153ad1ace3534f6dba68becfe69f418ce9d90e7b4ab3ff2f8e16a731R3-R5) and [lines 376 to 380](https://github.com/flutter/website/pull/9394/files#diff-afe162e1153ad1ace3534f6dba68becfe69f418ce9d90e7b4ab3ff2f8e16a731R376-R380). I don't need a second review and all comments are provided as GitHub suggestions for ease of review, acceptance, or both.

File: src/release/breaking-changes/3-13-deprecations.md
Comment (Piinks) [reply src/release/breaking-changes/3-13-deprecations.md:51]: This formatting is the same in all past migration guides, so I am going to leave it as is.

**Review by Piinks — COMMENTED** [review]
Thanks @atsansone! I left out changing heading formatting since it would probably mean going back to all migration guides and changing the heading formats. I don't have permission to merge, everything else has been updated. Thanks!

**Review by atsansone — APPROVED** [review]

File: src/release/breaking-changes/3-13-deprecations.md
Comment (felipecastrosales) [reply src/release/breaking-changes/3-13-deprecations.md]: this not follow this: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#use-correct-grammar 

or are there exception in this case, @atsansone ?

**Pull request merged**
